dashboard/modules/serve/serve_rest_api_impl.py

```diff
@@ -89,6 +89,7 @@ def __init__(self, dashboard_head_or_agent):
             # If the lock is already acquired by another async task, the async task
             # will asynchronously wait for the lock.
             self._controller_start_lock = asyncio.Lock()
+            self._session_name = dashboard_head_or_agent.session_name
```

Comment:
QQ: What's the purpose of adding this entry? Is it because we will have multiple dashboard sessions (Ray Dashboard, Train Dashboard)?

Reply:
if we tgrack train workers with their session name, we can use this to identify if this cluster has been restarted since that train worker was run. If so, we know not we cannot look it up in the train runs api.

Reply:
Ah I see. That makes sense.